components/calendar/src/cal/hijri.rs

```diff
@@ -1128,6 +1151,405 @@ impl Date<HijriUmmAlQura> {
+        } else {
+            let next_year = self.load_or_compute_info(extended_year + 1);
+            if rd < next_year.start_day {
+                year
+            } else {
+                next_year
+            }
+        };
+        let (m, d) = y.md_from_rd(rd);
+        HijriIranDateInner(ArithmeticDate::new_unchecked(y, m, d))
+    }
+
+    fn to_rata_die(&self, date: &Self::DateInner) -> RataDie {
+        date.0.year.md_to_rd(date.0.month, date.0.day)
+    }
+
+    fn from_iso(&self, iso: IsoDateInner) -> Self::DateInner {
+        self.from_rata_die(Iso.to_rata_die(&iso))
+    }
+
+    fn to_iso(&self, date: &Self::DateInner) -> IsoDateInner {
+        Iso.from_rata_die(self.to_rata_die(date))
+    }
+
+    fn months_in_year(&self, date: &Self::DateInner) -> u8 {
+        date.0.months_in_year()
+    }
+
+    fn days_in_year(&self, date: &Self::DateInner) -> u16 {
+        date.0.days_in_year()
+    }
+
+    fn days_in_month(&self, date: &Self::DateInner) -> u8 {
+        date.0.days_in_month()
+    }
+
+    fn offset_date(&self, date: &mut Self::DateInner, offset: DateDuration<Self>) {
+        date.0.offset_date(offset, &HijriIran)
+    }
+
+    fn until(
+        &self,
+        date1: &Self::DateInner,
+        date2: &Self::DateInner,
+        _calendar2: &Self,
+        _largest_unit: DateDurationUnit,
+        _smallest_unit: DateDurationUnit,
+    ) -> DateDuration<Self> {
+        date1.0.until(date2.0, _largest_unit, _smallest_unit)
+    }
+
+    fn debug_name(&self) -> &'static str {
+        Self::DEBUG_NAME
+    }
+
+    fn year_info(&self, date: &Self::DateInner) -> Self::Year {
+        era_year(self.extended_year(date))
+    }
+
+    fn extended_year(&self, date: &Self::DateInner) -> i32 {
+        date.0.extended_year()
+    }
+
+    fn is_in_leap_year(&self, date: &Self::DateInner) -> bool {
+        Self::provided_year_is_leap(date.0.year)
+    }
+
+    fn month(&self, date: &Self::DateInner) -> types::MonthInfo {
+        date.0.month()
+    }
+
+    fn day_of_month(&self, date: &Self::DateInner) -> types::DayOfMonth {
+        date.0.day_of_month()
+    }
+
+    fn day_of_year(&self, date: &Self::DateInner) -> types::DayOfYear {
+        date.0.day_of_year()
+    }
+
+    fn calendar_algorithm(&self) -> Option<crate::preferences::CalendarAlgorithm> {
+        let expected_calendar = crate::preferences::CalendarAlgorithm::Hijri(Some(
+            crate::preferences::HijriCalendarAlgorithm::HijriIran,
+        ));
+        Some(expected_calendar)
+    }
+}
+
+impl PrecomputedDataSource<HijriYearInfo> for HijriIran {
+    fn load_or_compute_info(&self, year: i32) -> HijriYearInfo {
+        #[rustfmt::skip]
+        const DATA: [PackedHijriYearInfo; 1447 - 1265] = {
+            use calendrical_calculations::iso::const_fixed_from_iso as iso;
+            let l = true; // long
+            let s = false; // short
+            // This uses the data gathered in https://github.com/roozbehp/qamari
+            [
```

Comment:
suggestion: this module is getting very large; we should have a separate private module that contains this data as constants.

Reply:
That's indeed understandable just that I followed what is done in UmmAlqura one https://github.com/unicode-org/icu4x/blob/24b7b162bdf6a473147c08803068f01cc82f5901/components/calendar/src/cal/hijri.rs#L799 so a change should be applied to that also I believe.

Reply:
So let's start with the UmAlQura one, https://github.com/unicode-org/icu4x/pull/6502

Reply:
Done https://github.com/unicode-org/icu4x/pull/6474/commits/2179833c600c8b6e13db2ec665f9bc049813ebbf